ops/framework.py

```diff
@@ -518,7 +518,10 @@ def __getattr__(self, name: str) -> BoundEvent:
 
 
 class LifecycleEvent(EventBase):
-    """Events tied to the lifecycle of the Framework object."""
+    """Events tied to the lifecycle of the Framework object.
+
+    Note: it is totally different from Juju [application] lifecycle.
```

Comment:
I'd prefer explicitly mentioning that we believe this was poorly-named, and agree with Tony about linking to the Juju docs. Something like this (but this uses Markdown links as I always forget the Sphinx linking syntax, so fix that ;-).

```suggestion
    Note: this was poorly-named. It has a different meaning than the
    [Juju lifecycle events](https://juju.is/docs/sdk/events) like "start" and "install".
```

(Except that

Reply:
added poorly named, disagree on external juju link.